### DISCUSSION TITLE
paste long text as file

### DISCUSSION THREAD

**Discussion by lucyknada** [open-webui/open-webui#6099] [Ideas]
# Feature Request

**Is your feature request related to a problem? Please describe.**
currently pasting very long text does not collapse it into a virtual file attached to the message, making it very messy to scroll back and forth to adjust the prompt

**Describe the solution you'd like**
claude and other frontends seem to detect a large paste and instead attach it as a file to the message, making prompting much easier, possibly even using a similar artifact slideout window to modify the attached files, but just attaching it as context would already be plenty nice as is

**Describe alternatives you've considered**
so far I've been pasting manually into a file and attaching it this way, but thats often very cumbersome

**Comment by taylorwilsdon** [comment]
Would love this, makes it much cleaner to interact with large code pastes - I'll take a run at it and cut a PR using the [existing paste handler](https://github.com/open-webui/open-webui/blob/main/src/lib/components/common/RichTextInput.svelte#L410)

**Reply by taylorwilsdon** [reply]
Hey @lucyknada - I have put up a fully functional PR for this @ https://github.com/open-webui/open-webui/pull/7020
If @tjbck thinks it makes sense, would love to get this merged!

**Reply by lucyknada** [reply]
that's exactly what I was hoping for, thanks so much!

**Reply by taylorwilsdon** [reply]
@lucyknada looks like it will ship with the coming 0.4 release! https://github.com/open-webui/open-webui/pull/6522/commits/11e78eac914c89cfbdbf2964aa1d0f8c90943a3a

**Reply by lucyknada** [reply]
exciting! thanks for the heads up and getting the ball rolling with the PR!

**Comment by taylorwilsdon** [comment]
Here's a video of it in action - 

https://github.com/user-attachments/assets/717f0bce-2362-4beb-a11c-851a0f49f36b

**Comment by lucyknada** [comment]
@taylorwilsdon does the feature work for you? I just pulled dev (though it's supposed to be in main too by now) and pasting 24k+ chars has no effect

**Reply by lucyknada** [reply]
found it; has to be toggled on in interface settings

**Comment by TSAGC** [comment]
Hi @lucyknada, would you be so kind as to help us. We're trying to enable "paste as file" for all our users as default, but we can't find the setting.

Could you let us know how to set this as the default for our instance? Thanks!

**Reply by Classic298** [reply]
settings > interface

to set as default for all you'll currently have to use this workaround https://github.com/open-webui/open-webui/pull/20770#issuecomment-3910169898

**Reply by TSAGC** [reply]
Thank you very much.
